src/Illuminate/Collections/Collection.php

```diff
@@ -1411,6 +1411,28 @@ public function transform(callable $callback)
         return $this;
     }
 
+    /**
+     * Return only unique items from the collection array.
+     *
+     * @param  string|callable|null  $key
+     * @param  bool  $strict
+     * @return static
+     */
+    public function unique($key = null, $strict = false)
+    {
+        $callback = $this->valueRetriever($key);
+
+        $exists = [];
+
+        return $this->reject(function ($item, $key) use ($callback, $strict, &$exists) {
+            if (in_array($id = $callback($item, $key), $exists, $strict)) {
+                return true;
+            }
+
+            $exists[] = $id;
+        });
+    }
```

Comment:
Just to explain why this didn't work:

The call to `$this->reject` returned a new `LazyCollection` instance. Every time the collection is enumerated, the callback (that was passed to `reject`) gets called again, but it's reusing the same `$exists` cache of found items, so anything that was previously enumerated is treated as a duplicate :frowning: 

The solution is to have a separate implementation of the `unique()` method for the `LazyCollection` class. In that implementation, we can move the `$exists` cache into the `Generator` function passed to the constructor, so that each enumeration has its own cache :+1: